src/core/Spy.js

```diff
@@ -0,0 +1,58 @@
+getJasmineRequireObj().Spy = function (j$) {
+
+  function Spy(name, originalFn) {
+    var args = buildArgs(),
+      /*`eval` is the only option to preserve both this and context:
+        - former is needed to work as expected with methods,
+        - latter is needed to access real spy function and allows to reduce eval'ed code to absolute minimum
+        More explanation here (look at comments): http://www.bennadel.com/blog/1909-javascript-function-constructor-does-not-create-a-closure.htm
+       */
+      /* jshint evil: true */
+      wrapper = eval('(function (' + args + ') { return spy.apply(this, Array.prototype.slice.call(arguments)); })'),
```

Comment:
I was trying with something like that before I used an `eval`. It looks that when function is created with `new Function` constructor, then it losts all of context in which it was created. So then I tried passing a `spy` as first argument, but then I went into troubles with preserving `this` which resulted the line above.